ReadingList.Notes.Github/Services/GithubBookRecordService.cs

```diff
@@ -47,6 +48,23 @@ public async Task<IEnumerable<BookRecord>> GetAllBookRecords()
             }
         }
 
+        public async Task<BookRecord> GetBookNotes(string book)
+        {
+            try
+            {
+                var repo = await _githubFileClient.GetRepositoryContent(UserName, Repo, NotesFolder);
+                
+                var bookContent = repo?.FirstOrDefault(b => b.Name.ToLower().Contains(book.ToLower()));
+                return await GetBookRecord(bookContent);
+            }
+            catch (RateLimitExceededException rateLimitException)
+            {
+                _logger.Error("Rate limit exceeded against Github, falling back to cache: ", rateLimitException);
+                return _bookRecordCache.GetAll()
```

Comment:
Should consider splitting up to it's own models here so we can have the SHA-hash and raw filename without bleeding out to the external API.